### ISSUE TITLE
`while_let_on_iterator` false positive with nested loop

### ISSUE THREAD

**Issue by sinkuu** [Manishearth/rust-clippy#1438]
```rust
fn main() {
    let mut it = 1..100;

    for _ in 0..10 { // or any other loop
        while let Some(x) = it.next() { // (false positive) this loop could be written as a `for` loop
        }
    }
}

```

`while_let_on_iterator` lint should ignore a `while let Some(x) = iterator_variable_external_to_loop` within a loop, because iterators cannot be consumed there.

**Comment by sinkuu** [comment]
Duplication of https://github.com/Manishearth/rust-clippy/issues/600
